File: cinema/src/main/java/com/example/cinema/seat/Seat.java
package com.example.cinema.seat;

import com.example.cinema.room.Room;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import static jakarta.persistence.FetchType.EAGER;
import static jakarta.persistence.FetchType.LAZY;

@Entity
@Setter
@Getter
public class Seat {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private char seatRow;
    private int number;
    @ManyToOne(fetch = EAGER)
    Room room;
}

File: cinema/angularclient/src/app/model/seat.ts
import { Room } from "./room";

export class Seat {
    id: number;
    seatRow: string;
    number: number;
    room: Room;

    constructor() {
        this.id = 0
        this.seatRow = "";
        this.number = 0;
        this.room= new Room();
    }
}
